src/lib.rs

```diff
@@ -1351,6 +1351,58 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 		Ok(())
 	}
 
+	/// Sends payment probes over all paths of a route that would be used to pay the given
+	/// amount to the given `node_id`.
+	///
+	/// This may be used to send "pre-flight" probes, i.e., to train our scorer before conducting
+	/// the actual payment.
+	pub fn send_spontaneous_payment_probe(
```

Comment:
Could you DRY up this and `send_payment_probe`? Looks like only constructing `RouteParameters` is different.